flax/experimental/nnx/nnx/flaglib.py

```diff
@@ -60,7 +63,15 @@ def __call__(self, **kwargs: tp.Any):
     finally:
       FLAGS_CONTEXT.flags_stack.pop()
 
-  def get(self, name: str, default: tp.Any = None) -> tp.Optional[tp.Any]:
+  @tp.overload
+  def get(self, name: str) -> tp.Any:
+    ...
+
+  @tp.overload
+  def get(self, name: str, default: A) -> A:
+    ...
+
+  def get(self, name: str, default: A = None) -> A | None:
```

Comment:
do you need a separate `overload` if the method signatures are the same?

Reply:
The last non-@overloaded definition is not used by type checkers, so this LGTM.